File: medium/4. Monotonic Array/monotonic_array.cpp
#include <vector>

bool isMonotonic(std::vector<int> array) {
    int n = array.size();
    if (n <= 2) return true;
    bool is_increasing = true;
    bool is_decreasing = true;
    for (int i = 1; i < n; i++) {
        if (array[i] > array[i - 1]) {
            is_decreasing = false;
        } else if (array[i] < array[i - 1]) {
            is_increasing = false;
        }
    }
    return is_increasing || is_decreasing;
}

File: dsa/medium/4. Monotonic Array/monotonic_array.go
package main

func IsMonotonic(array []int) bool {
	if len(array) < 2 {
		return true
	}
	a_len := len(array)
	is_descreasing := true
	is_increasing := true

	for i := 1; i < a_len; i++ {
		if array[i] > array[i-1] {
			is_descreasing = false
		}
		if array[i] < array[i-1] {
			is_increasing = false
		}
	}
	return is_descreasing || is_increasing
}
